lib/net.js

```diff
@@ -1416,7 +1416,7 @@ function lookupAndConnect(self, options) {
         // calls net.Socket.connect() on it (that's us). There are no event
         // listeners registered yet so defer the error event to the next tick.
         process.nextTick(connectErrorNT, self, err);
-      } else if (!isIP(ip)) {
+      } else if ((typeof ip !== 'string') || !isIP(ip)) {
```

Comment:
Why don't we handle this inside isIp() function?

Reply:
You have a point. I think this is reasonable to have this inside of `isIP`. Let me check.

Reply:
`isIP` is a public API. It would be a breaking change to do it.